### PR TITLE
If image bucket is unavailable, download svg file

### PR DESCRIPTION
**Pull request by johnhyde**
This opens up the "Export canvas" to everyone.

In the future, perhaps we should render it to a PNG, as most image hosting sites like imgur don't accept SVGs.

### CODE REVIEW

**Comment by johnhyde** [comment]
Ok, I've added PNG export as a menu option

**Comment by yosoyubik** [comment]
Downloading SVG and PNG files works fine but If an S3 bucket is available, exporting to PNG exports [this](https://yosoyubik.fra1.digitaloceanspaces.com/zod/~2021.12.11..06.57.42-hom.png)? Not sure what

File: ui/src/components/OptionsMenu.svelte [hunk 1]
```diff
@@ -76,6 +54,73 @@
...
+    if (canUploadFilesToStorage()) {
+      try {
+        await uploadFileToStorage(svgData, fileName, mimetype);
+        return;
+      } catch (e) {
+        console.error(`Encountered error uploading to s3: ${e.message}\nDownloading svg file`, e);
+      }
+    }
+    downloadFile(dataUrl, fileName);
+  }
```
Comment (yosoyubik) [inline ui/src/components/OptionsMenu.svelte:123]:
```suggestion
    switch (fileExtension) {
      case 'svg':
        data = svgData;
        dataUrl = generateDataUrl(svgData, 'image/svg+xml');
        mimetype = 'image/svg+xml';
        break;
      case 'png':
        ({ dataUrl } = await renderSvgToPng(svgData));
        data = dataUrl;
        mimetype = 'image/png';
        break;
      default:
        throw new Error(`Unknown file extension: ${fileExtension}`);
    }
    const timestamp = dateToDa(new Date());
    const fileName = `${window.ship}/${timestamp}-${name}.${fileExtension}`;
    if (canUploadFilesToStorage()) {
      try {
        await uploadFileToStorage(data, fileName, mimetype);
      } catch (e) {
        console.error(
          `Encountered error uploading to s3: ${e.message}\nDownloading svg file`,
          e
        );
      }
    } else {
      downloadFile(dataUrl, fileName);
    }
```

File: ui/src/components/OptionsMenu.svelte [hunk 2]
```diff
@@ -76,6 +54,73 @@
...
+    }
+
+    const bucket = $store.s3.configuration.currentBucket;
+    const params = new PutObjectCommand({
+      Bucket: bucket,
+      Key: fileName,
+      Body: data,
+      ACL: 'public-read',
+      ContentType: mimetype
+    });
```
Comment (yosoyubik) [inline ui/src/components/OptionsMenu.svelte:74]:
```suggestion
    const body =
      mimetype !== 'image/png'
        ? data
        : Buffer.from(data.replace(/^data:image\/\w+;base64,/, ''), 'base64');

    const bucket = $store.s3.configuration.currentBucket;
    const params = new PutObjectCommand({
      Bucket: bucket,
      Key: fileName,
      Body: body,
      ACL: 'public-read',
      ContentEncoding: mimetype === 'image/png' ? 'base64' : undefined,
      ContentType: mimetype
    });
```

**Review by yosoyubik — CHANGES_REQUESTED** [review]
I got it to work with the suggested changes I posted—let me know what you think

**Review by yosoyubik — CHANGES_REQUESTED** [review]

**Comment by johnhyde** [comment]
Changes implemented

**Pull request merged**